app/controllers/postgresql_lo_streamer/lo_controller.rb

```diff
@@ -2,78 +2,27 @@
 
 module PostgresqlLoStreamer
   class LoController < ActionController::Base
+
     def stream
-      send_file_headers!(headers_for_extension_or_default)
 
-      object_identifier = params[:id].to_i
-      if !object_exists?(object_identifier)
-        self.status = 404
-        render :nothing => true and return
+      streamer = PostgresqlLoStreamer::Streamer.new(
+        connection,
+        params[:id].to_i,
+        params[:format]
```

Comment:
The streamer shouldn't know anything about the format. It only knows how to stream a binary blob from the database. It's up to the controller to decide on how to interpret the data.